CMakeLists.txt

```diff
@@ -233,6 +233,8 @@ if (GCC)
   # https://cmake.org/cmake/help/latest/command/try_run.html
   set(MEMCMP_INVALID_STRIPPED "")
   set(MEMCMP_INVALID_STRIPPED__TRYRUN_OUTPUT "")
+  # See https://gitlab.kitware.com/cmake/cmake/-/issues/16920 for why we set this to 2
+  set(THREADS_PTHREAD_ARG "2" CACHE STRING "Forcibly set by CMakeLists.txt." FORCE)
```

Comment:
which command in our cmake is creating the issue that is solved by this line?

Reply:
The issue is the try run command here https://github.com/awslabs/aws-lc/blob/main/CMakeLists.txt#L207 . If we try to find-threads then it will run a test program for the threading library and the test program apparently needs the `THREADS_PTHREAD_ARG` set to 2 otherwise it will fail.

Reply:
ok, can you please add that to the comment and open a ticket for fixing this?

Reply:
I don't really understand what we would fix here, the try_run command is part of cmake, and this appears to be undocumented (as far as I know) behavior for try_run. Our options are to either set the argument here in order to make sure this works for cross-compiled code, or stop using try_run entirely. The info I put there was from the issue I linked above and the comments on it.

Reply:
yeah, I meant we can maybe stop using try_run and find another solution for that.